commons-rng-sampling/src/main/java/org/apache/commons/rng/sampling/distribution/ChengBetaSampler.java

```diff
@@ -266,6 +266,7 @@ public double sample() {
                 // Compute Y and Z
                 final double y = u1 * u2;
                 final double z = u1 * y;
+                final double v1 = Math.log(u1) - Math.log1p(-u1);
```

Comment:
Maybe another name should be better, but it's monday and i'm not so creative

Reply:
If you look at where this computation is done it is not always required in each iteration of the loop. Only step 3 and step 5 require it, and not both at the same time. 50% of the time step 2 is executed and you do not require this value. It is more efficient to compute it when required. So this change will slow down the sampler with 2 unnecessary calls to log functions in the majority of cases.